tests/js/session-tests.js

```diff
@@ -1152,6 +1152,225 @@ module.exports = {
+    realm1.write(() => {
+      for (let i = 0; i < 25; i++) {
+        realm1.create("Person", {
+          _id: new ObjectId(),
+          age: i,
+          firstName: "John",
+          lastName: "Smith",
+        });
+      }
+    });
+
+    // we haven't uploaded our recent changes -- we're not allowed to copy
+    TestCase.assertThrowsContaining(() => {
+      realm1.writeCopyTo(realm1Path + "copy2.realm");
+    }, "Could not write file as not all client changes are integrated in server");
+
+    // log back in and upload the changes we made locally
+    user1 = await app.logIn(credentials1);
+    await realm1.syncSession.uploadAllLocalChanges();
+
+    // create copy no. 2 of the realm
+    const realm2Path = realm1Path + "copy2.realm";
+    realm1.writeCopyTo(realm2Path);
+
+    /*
+        Test 2:  check that a copied realm can be opened by another user, and that
+          the contents of the original realm and the copy are as expected
+      */
+    // log in a new user, open the realm copy we created just above
+    const user2 = await app.logIn(credentials2);
+    const config2 = {
+      sync: {
+        user: user2,
+        partitionValue: partition,
+        _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+      },
+      schema: [schemas.PersonForSync, schemas.DogForSync],
+      path: realm2Path,
+    };
+
+    const realm2 = await Realm.open(config2);
+
+    let realm1Persons = realm1.objects("Person");
+    let realm2Persons = realm2.objects("Person");
+    TestCase.assertEqual(
+      // the contents of the two realm should be the same
+      realm1Persons.length,
+      realm2Persons.length,
+      "The same number of people should be in the two realms",
+    );
+
+    // add another 25 people locally to the original realm
+    realm1.syncSession.pause();
+    realm1.write(() => {
+      for (let i = 0; i < 25; i++) {
+        realm1.create("Person", {
+          _id: new ObjectId(),
+          age: i,
+          firstName: "John",
+          lastName: "Smith",
+        });
+      }
+    });
+
+    realm1Persons = realm1.objects("Person");
+    realm2Persons = realm2.objects("Person");
+    TestCase.assertTrue(
+      // realm1 and realm2 should no longer be the same
+      realm1Persons.length == realm2Persons.length + 25,
+      "realm1 should have an additional 25 people",
+    );
+    realm1.syncSession.resume();
+
+    await realm1.syncSession.uploadAllLocalChanges();
+    await realm1.syncSession.downloadAllServerChanges();
+
+    await user2.logOut();
+    realm2.close();
+    Realm.deleteFile(config2);
+    const realm3Path = realm1Path + "copy3.realm";
+    realm1.writeCopyTo(realm3Path);
+
+    /*
+      Test 3:  open a copy of our realm with a new user and a new
+        partition key.  We expect it to fail because of the mismatch
+        in partition keys
+    */
+    const user3 = await app.logIn(credentials3);
+    const otherPartition = Utils.genPartition();
```

Comment:
```suggestion

    /*
      Test 3:  open a copy of our realm with a new user and a new
        partition key.  We expect it to fail because of the mismatch
        in partition keys
    */
    const realm3Path = realm1Path + "copy3.realm";
    realm1.writeCopyTo(realm3Path);

    const user3 = await app.logIn(credentials3);
    const otherPartition = Utils.genPartition();
```

Maybe slightly clearer if this action comes after the description?

Reply:
👍